internal/controller/repository/repository.go

```diff
@@ -202,6 +213,73 @@ func getUserPermissionMapFromCr(users []v1alpha1.RepositoryUser) map[string]stri
 	return crMToPermission
 }
 
+func getWebhookMapFromCr(webhooks []v1alpha1.RepositoryWebhook) map[string]map[string]interface{} {
+	crWToConfig := make(map[string]map[string]interface{}, len(webhooks))
+
+	for _, webhook := range webhooks {
+		sort.Strings(webhook.Events)
+		crWToConfig[webhook.Url] = map[string]interface{}{
+			"content_type": webhook.ContentType,
+			"events":       webhook.Events,
+		}
+	}
+	return crWToConfig
+}
+
+func getRepoWebhooksWithConfig(ctx context.Context, gh *ghclient.Client, org, name string) (map[string]map[string]interface{}, error) {
+	wToConfig := make(map[string]map[string]interface{})
+
+	opt := &github.ListOptions{PerPage: 100}
+
+	for {
+		hooks, resp, err := gh.Repositories.ListHooks(ctx, org, name, opt)
```

Comment:
function could accept a list of hooks to reduce api-calls

Reply:
Working on it

Reply:
Implemented.